src/Components/List/index.jsx

```diff
@@ -0,0 +1,47 @@
+import { Pagination } from '@mantine/core'
+import { useContext, useState } from "react"
+import { SettingsContext } from "../../Context/settings"
+
+export const List = () => {
+  const {
+    list, setList,
+    itemsPerPage,
+    showComplete
+  } = useContext(SettingsContext)
+  const [page, setPage] = useState(1);
+
+  function deleteItem(id) {
+    const items = list.filter(item => item.id !== id);
+    setList(items);
+  }
+
+  function toggleComplete(id) {
+    const items = list.map(item => {
+      if (item.id === id) {
+        item.complete = !item.complete;
```

Comment:
Better to return a toggled copy of `item` here, rather than modifying `item` in-place.

Reply:
I'm not quite sure what you mean on this one.

Reply:
Instead of modifying the object in place, return a new object in its place like so:
```
list.map((item) => {
    if (item.id === id) {
        return {
            ...item,
            complete: !item.complete
        };
    }

   return item;
});
```
As a side note: Always avoid setting state objects directly, even if you pass your new state to the setter function, if the refs are the same it will not trigger a rerender... which will be a pita to debug 😉 you're safe here, fortunately, since you're using map which creates a new array, but I thought I'd throw it out there for future reference